CMakeLists.txt

```diff
@@ -159,6 +159,17 @@ if (ONEDPL_BACKEND MATCHES "^(tbb|dpcpp|dpcpp_only)$")
                 "Select one of the following devices: CPU, GPU, FPGA_EMU or FPGA_HW")
         endif()
 
+        # check device backend
+        if (ONEDPL_DEVICE_BACKEND AND ONEDPL_DEVICE_BACKEND MATCHES "^(opencl|level_zero?)$")
```

Comment:
Part `ONEDPL_DEVICE_BACKEND AND` looks useless.
Could you please explain the regexp? I don't quite understand `?` and `*` case is not covered but declared in the error message below.

Consider: `^(opencl|level_zero|\*)$`